### PR TITLE
docs: allow the chaining of configs

### PR DESCRIPTION
**Pull request by crawford**
This describes how the config can either append multiple configs together or
replace itself altogether with another config.

Builds on https://github.com/coreos/ignition/pull/165.

### CODE REVIEW

File: doc/configuration.md [hunk 1]
```diff
@@ -2,7 +2,48 @@
...
 The Ignition configuration is a JSON document conforming to the following specification, with **_italicized_** entries being optional:
 
-* **ignitionVersion** (integer): the version number of the spec. Must be `1`.
+* **ignition** (object): metadata about the configuration itself.
+  * **version** (string): the semantic version number of the spec. Must be at least `2.0.0`.
+  * **_append_** (object): append one or more configs to the current config.
+    * **_configs_** (list of objects): a list of the configs to be appended.
+      * **source** (object): the source of the config.
+        * **uri** (string): the URI of the remote config.
+      * **_verification_** (object): options related to the verification of the config.
```
Comment (jonboulle) [inline doc/configuration.md]: On the one hand, I appreciate the flexibility/future proofing; on the other hand, the nesting is awfully hairy. I wonder if we can be more opinionated, with more assumptions (e.g., only support detached signatures), and consequently be able to flatten and simplify this schema a bit.

Comment (crawford) [inline doc/configuration.md]: I don't think I like the idea of flattening the config. But we could definitely _pare_ down the config to only the options we are planning on supporting for now.

Comment (crawford) [inline doc/configuration.md]: I'm also opting for an overly verbose structure to as to match other similar options. For example, if you look at https://github.com/crawford/ignition/commit/da52c35b5ec828ffcd2b8a1518087c033f40f68c, you'll see `contents.source.uri` and `contents.source.immediate`. I'm using that same structure here, even though only `configs.source.uri` exists. This is just to help reduce the cognitive overhead of writing correct configs. Thooooough, if we are telling people that humans shouldn't write the configs, what does it matter?

**Comment by mischief** [comment]
if you really don't expect hoomans to write ignition, why bother with immediate/uri? just parse [data url schemes](http://tools.ietf.org/html/rfc2397) in uri.

**Comment by crawford** [comment]
@mischief good idea.

**Comment by mischief** [comment]
@crawford really, you could replace nearly any field that takes data (say for example, systemd unit file content, file content) with uri and accept data uri. but, maybe that's too much of a pain. (think uri encoding all these fields, like space/nl in systemd unit files).

but, as i said, if ignition.json really is a low-level format, that won't matter. we just need to make tools available to ease authoring, editing, parsing, viewing, etc.

**Comment by crawford** [comment]
@jonboulle thanks to @mischief's suggestion, I was able to flatten this quite a bit. I also took a page out of AppC with regard to the hashes.

**Comment by crawford** [comment]
Pruned down and implemented in https://github.com/coreos/ignition/pull/183.

**Pull request closed without merging**